Latest change result in dll.h not found error

With recent changes , da1c8d360e6b9ba8f1bc291728263548a8e30698, removing dll.h from the source code and using generate_export_header to generate dll.h is not working as it fails to generate dll.h and I am erroring out with errors in my main CMakeList with,
 fatal error: 'yaml-cpp/dll.h' file not found
#include "yaml-cpp/dll.h"

Context: I am using yaml-cpp as external project via ExternalProject_Add(). Earlier dll.h was part of source tree in include/yaml-cpp/dll.h and references in the my own client code pointed to #include<yaml-cpp/dll.h> . My include_directories() was having yaml-src/include. And all worked well.

With latest change, the dll.h will be created in build/include/yaml-cpp folder. This breaks my code build unless I add build/include/yaml-cpp to include_directories().
It results in errors like ,
include/yaml-cpp/node/node.h:13:10: fatal error: 'yaml-cpp/dll.h' file not found
 #include "yaml-cpp/dll.h"

PS: Adding build/include/yaml-cpp to include_directories() fixes the error. Is this right enough way?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,11 +14,6 @@ include(CMakeDependentOption)
 include(CheckCXXCompilerFlag)
 include(GNUInstallDirs)
 include(CTest)
-include(GenerateExportHeader)
-
-set(CMAKE_C_VISIBILITY_PRESET hidden)
-set(CMAKE_CXX_VISIBILITY_PRESET hidden)
-set(CMAKE_VISIBILITY_INLINES_HIDDEN 1)
 
 find_program(YAML_CPP_CLANG_FORMAT_EXE NAMES clang-format)
 
@@ -86,7 +81,6 @@ set_property(TARGET yaml-cpp
 target_include_directories(yaml-cpp
   PUBLIC
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
-    $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/include>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   PRIVATE
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
@@ -142,12 +136,6 @@ write_basic_package_version_file(
   "${PROJECT_BINARY_DIR}/yaml-cpp-config-version.cmake"
   COMPATIBILITY AnyNewerVersion)
 
-generate_export_header(yaml-cpp
-  BASE_NAME YAML_CPP
-  EXPORT_FILE_NAME "${PROJECT_BINARY_DIR}/include/yaml-cpp/dll.h"
-  EXPORT_MACRO_NAME YAML_CPP_API
-)
-
 configure_file(yaml-cpp.pc.in yaml-cpp.pc @ONLY)
 
 if (YAML_CPP_INSTALL)
@@ -157,9 +145,6 @@ if (YAML_CPP_INSTALL)
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
     ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 	install(DIRECTORY ${PROJECT_SOURCE_DIR}/include/
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
-		FILES_MATCHING PATTERN "*.h")
-	install(DIRECTORY ${PROJECT_BINARY_DIR}/include/
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 		FILES_MATCHING PATTERN "*.h")
   install(EXPORT yaml-cpp-targets

--- a/include/yaml-cpp/dll.h
+++ b/include/yaml-cpp/dll.h
@@ -1,0 +1,33 @@
+#ifndef DLL_H_62B23520_7C8E_11DE_8A39_0800200C9A66
+#define DLL_H_62B23520_7C8E_11DE_8A39_0800200C9A66
+
+#if defined(_MSC_VER) ||                                            \
+    (defined(__GNUC__) && (__GNUC__ == 3 && __GNUC_MINOR__ >= 4) || \
+     (__GNUC__ >= 4))  // GCC supports "pragma once" correctly since 3.4
+#pragma once
+#endif
+
+// The following ifdef block is the standard way of creating macros which make
+// exporting from a DLL simpler. All files within this DLL are compiled with the
+// yaml_cpp_EXPORTS symbol defined on the command line. This symbol should not
+// be defined on any project that uses this DLL. This way any other project
+// whose source files include this file see YAML_CPP_API functions as being
+// imported from a DLL, whereas this DLL sees symbols defined with this macro as
+// being exported.
+#undef YAML_CPP_API
+
+#ifdef YAML_CPP_DLL      // Using or Building YAML-CPP DLL (definition defined
+                         // manually)
+#ifdef yaml_cpp_EXPORTS  // Building YAML-CPP DLL (definition created by CMake
+                         // or defined manually)
+//	#pragma message( "Defining YAML_CPP_API for DLL export" )
+#define YAML_CPP_API __declspec(dllexport)
+#else  // yaml_cpp_EXPORTS
+//	#pragma message( "Defining YAML_CPP_API for DLL import" )
+#define YAML_CPP_API __declspec(dllimport)
+#endif  // yaml_cpp_EXPORTS
+#else   // YAML_CPP_DLL
+#define YAML_CPP_API
+#endif  // YAML_CPP_DLL
+
+#endif  // DLL_H_62B23520_7C8E_11DE_8A39_0800200C9A66

--- a/src/node_data.cpp
+++ b/src/node_data.cpp
@@ -13,7 +13,7 @@
 
 namespace YAML {
 namespace detail {
-YAML_CPP_API std::atomic<size_t> node::m_amount{0};
+std::atomic<size_t> node::m_amount{0};
 
 const std::string& node_data::empty_scalar() {
   static const std::string svalue;

